### ISSUE TITLE
Refunds / Returns not shown in order history

### ISSUE THREAD

**Issue by giacogiaco** [woocommerce/woocommerce-admin#6693]
**Describe the bug**
Refunded orders are not listed in the order history for that day. 

**To Reproduce**
1. Go to Analytics > Revenue
2. Click on the section 'RETURNS'
3. In the graph, find a day on which a return happened
4. Then, in the table below, find that day and (in column two) click on the orders of that day
5. You'll be lead to a page that shows you all orders of that specific day – unfortunately, it doesn't show the order that was refunded. Why not?

**Expected behavior**
There should be some indication of a refund that happened on that day. Maybe link to the day on which the order that was later refunded was placed.

**Additional context**
Is **returns** the same as **refunds**?

**Comment by joshuatf** [comment]
Hi @giacogiaco - thanks for opening this issue.

I believe I understand the issue correctly.  You're referring to clicking on the order count for a given day in the revenue report that had a refund which then does not show you any items, correct?

<img width="758" alt="Screen Shot 2021-03-30 at 5 54 53 PM" src="https://user-images.githubusercontent.com/10561050/113062123-3f13c100-9181-11eb-8b5b-e88bbfba14ea.png">

I believe this is the currently expected behavior.  That page is meant to show orders that happened on a given date and not refunds.  

Although this is a change that would probably need to happen upstream in WooCommerce, I'm curious if you'd expect to see orders on this page as well.

**Comment by giacogiaco** [comment]
Hi @joshuatf, this problem is tied to https://github.com/woocommerce/woocommerce-admin/issues/6692. 

**You're referring to clicking on the order count for a given day in the revenue report that had a refund which then does not show you any items, correct?**
Correct. It is super confusing: There was an order (59€) on Feb 13. This order was refunded on Feb 18. 
- For Feb 13, the analytics graph it shows this order (among others) and it says refund 0€. Then, when opening the transaction view for that day (.../wp-admin/edit.php?post_type=shop_order&m=20210213), it shows all orders for that day and shows that one order was refunded. So why not show the refund (-59€) in the graph and analytics view too? 
- For Feb 18, in the analytics graph it shows the refund (-59€). But when opening the transaction view, it shows no transaction. Refunds are transactions though. 

This should be consistent. Refunds should be subtracted from the revenue (on the day the refunded revenue was generated).

**Comment by joshuatf** [comment]
>Correct. It is super confusing: There was an order (59€) on Feb 13. This order was refunded on Feb 18.

I agree that this is a bit confusing.  We'd have to update the semantics as well as the WooCommerce core functionality since this is specifically looking for orders on that date.

The discrepancy in transactions vs orders on a given date (and the fact that WooCommerce core shows the final status on an order instead of purely what happened on a given date) makes the difference in analytics and what core shows a bit a jarring.

>For Feb 18, in the analytics graph it shows the refund (-59€). But when opening the transaction view, it shows no transaction. Refunds are transactions though.

When you say "transaction view," are you referring to the WooCommerce orders page?  These are not technically orders, so this is what I'm referring to when I say the semantics would need to be updated and we would need a new page to show transactions in order to do this.  Refunds are not orders by themselves, but a transactions page in WooCommerce would be helpful.

>Refunds should be subtracted from the revenue (on the day the refunded revenue was generated).

Do you mean refunds should reduce the original order revenue on that order's original date in analytics?  If so, I don't believe this is necessarily true. Most businesses will operate on a cash basis and recording a credit/debit on the day a transaction occurred is usually the accepted method. cc @LevinMedia @timmyc for a sanity check on this.

That said, the tax issue you bring up in https://github.com/woocommerce/woocommerce-admin/issues/6692 is an interesting one that breaks this mold a bit.  I'll reply there with more thoughts on this.

**Comment by LevinMedia** [comment]
@giacogiaco thanks for pointing out this issue - I think I understand the problem you're facing too, although I'm not sure exactly how we'd go about solving it. Correct me if I'm wrong, but it sounds like you're looking for an easy way to say  "show me all the orders that had returns on this specific day". Correct? We might be able to solve this with advanced filters in the orders report, but I'll have to test it out. 

@joshuatf you are correct - In the analytics we are tracking returns (refunds) on the day they occur, rather than on the day the order / sale was originally recorded. This is common practice in business accounting. As an example, today assuming you had no sales, but you had a single return, you'd record a net loss for the day.

**Comment by LevinMedia** [comment]
>it sounds like you're looking for an easy way to say "show me all the orders that had returns on this specific day"

@giacogiaco if this is the use case you're looking for this can be achieved by first looking at the revenue report to figure out what day you want to see orders with refunds. Here's an example from my test store - with refunds from today's date, the 12th:

<img width="2652" alt="Screen Shot 2021-05-12 at 12 51 52 PM" src="https://user-images.githubusercontent.com/4500952/118036580-97fa7b80-b321-11eb-9044-4b7fb83c6019.png">

Once you know the date or range of days you want to look at, switch over to the orders report, and change the date filter so it's only showing the single day you want to see the refunds from. (in this case since it's today, I used the quick filter, but if you want to pick a single date from the past, you can do that using the custom date picker and selecting the date you want for both the start and finish of the date range) Once you have the date set add a "refund" advanced filter - and make sure it's set to all. 

You'll see a list of orders that had refunds on that day, regardless of the original date of the order. From there you can click the individual order number and inspect the order details. Here's the two orders that were refunded today, the 12th, even though the orders were created months ago. 

<img width="2652" alt="Screen Shot 2021-05-12 at 12 54 06 PM" src="https://user-images.githubusercontent.com/4500952/118036519-85804200-b321-11eb-9bfa-7f0f67e185de.png">

**Comment by dani-elo** [comment]
Hey @LevinMedia ,

Thanks for that solution. 

I found a similar issue in 4025488-zen and I thought that maybe a good solution would be to change the link from the `Revenue report` and make it point to the `Orders report` instead of the WooCommerce Orders list  (`WooCommerce > Orders > Filter by day`)

**Comment by stephhjacq** [comment]
Also reported in 4043125-zen

Thanks @LevinMedia your suggestions helped us get to the bottom of it. 

Per @dani-elo's suggestion, if the report linked to the filtered orders list this would be much easier to find.

**Comment by LevinMedia** [comment]
>I found a similar issue in 4025488-zen and I thought that maybe a good solution would be to change the link from the Revenue report and make it point to the Orders report instead of the WooCommerce Orders list (WooCommerce > Orders > Filter by day)

Generally I'm okay with this approach. My hesitation initially was we'd be changing the date of the reports through a new mechanism, rather than the filter at the top of the page. Normally when you move between reports, the date range persists, but in this case the date range would be filtered to the date of the orders or refunds of the row that was clicked. That said, I'm okay with change in behavior since the click action is originating in a row that's tied to a specific date. 

In addition to changing the link from the orders list to the orders report, filtered by the date of the row that was clicked, we should also link the refund amount directly to the order report, filtered by the day, with the refund filter applied, and refunds set to "all"

Revenue report:

<img width="1666" alt="Screen Shot 2021-06-17 at 7 40 14 AM" src="https://user-images.githubusercontent.com/4500952/122419790-08f40b00-cf40-11eb-9879-fbb53157da90.png">

Orders report:

<img width="1666" alt="Screen Shot 2021-06-17 at 7 40 46 AM" src="https://user-images.githubusercontent.com/4500952/122419933-245f1600-cf40-11eb-9e1f-b28ef561a7c4.png">

**Comment by joshuatf** [comment]
@LevinMedia Do you have a preferred way of handling this?

**Comment by mattsherman** [comment]
@LevinMedia Cycling back on this one to see if you have any input to provide.

**Comment by LevinMedia** [comment]
@joshuatf @mattsherman can we proceed with the plan I detailed in my most recent [comment](https://github.com/woocommerce/woocommerce-admin/issues/6693#issuecomment-863303812)?

**Comment by joshuatf** [comment]
Sounds like a good plan to me.  Thanks for confirming, @LevinMedia!
